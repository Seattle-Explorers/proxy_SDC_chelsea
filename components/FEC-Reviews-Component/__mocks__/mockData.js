module.exports = {
  "reviews": [
    {
      "reviewer_name": "Clifton",
      "dp": "25",
      "body": "Cillum magna cupidatat id veniam eiusmod et fugiat duis labore ipsum adipisicing cillum. Commodo consectetur et amet ipsum et anim aute commodo. Quis aliquip nisi ad ex. Tempor elit esse in reprehenderit eu proident nisi qui aute anim anim dolor dolore.",
      "date": "2015-04-18T07:58:27.719Z"
    },
    {
      "reviewer_name": "Alvis",
      "dp": "9",
      "body": "Tempor do laborum excepteur exercitation ipsum sunt non duis ad culpa minim nisi elit esse laboris. Non in pariatur cupidatat elit. Ullamco ad ut tempor excepteur reprehenderit enim proident sit officia esse mollit. Incididunt labore sunt consectetur qui aliqua sunt amet non eu excepteur aute sunt minim esse laborum. Qui excepteur proident pariatur consectetur velit. Ut nisi voluptate id nisi aliquip ad.",
      "date": "2016-04-08T06:41:51.623Z"
    },
    {
      "reviewer_name": "Samir",
      "dp": "10",
      "body": "Cupidatat voluptate aliqua veniam ullamco ut officia et velit dolor.",
      "date": "2016-09-08T08:28:25.808Z",
      "respond": "Aliqua laborum sint enim occaecat adipisicing magna laboris non tempor consectetur incididunt. Duis culpa quis do nisi eiusmod irure id aute voluptate laborum nisi eiusmod est culpa."
    },
    {
      "reviewer_name": "Ismael",
      "dp": "20",
      "body": "Est ut anim ipsum enim mollit excepteur sit. Sunt irure esse adipisicing tempor exercitation eu ex dolore sunt do officia pariatur. Dolor enim incididunt aute consectetur nisi ullamco quis. Eiusmod eiusmod velit excepteur non ut pariatur. Non deserunt aliqua do elit reprehenderit consectetur nisi est voluptate. Elit voluptate magna dolor qui ut amet fugiat nostrud deserunt anim Lorem aliqua cupidatat. Consequat in cupidatat dolor sunt culpa non esse id cillum laborum ut minim Lorem nostrud labore.",
      "date": "2017-08-17T13:43:48.413Z",
      "respond": "Eu tempor commodo ut nisi nulla ad ex. Aliquip nostrud minim cillum consequat. Duis aliquip aliquip proident non cillum deserunt laboris sit incididunt. Ea labore exercitation occaecat ipsum id nulla ipsum eu nostrud deserunt. Ut commodo velit cupidatat et nisi mollit aliqua adipisicing amet exercitation cillum nostrud ut. Ut Lorem culpa minim quis et. Excepteur veniam laboris sit mollit consequat sint ad. Magna dolor dolor ullamco officia ex duis aliquip aute ullamco quis."
    },
    {
      "reviewer_name": "Kolby",
      "dp": "16",
      "body": "Anim minim qui culpa anim. Esse exercitation sit sint tempor aliqua proident. Ullamco duis tempor cupidatat sunt quis labore mollit tempor. Ea enim aliquip Lorem do aliquip exercitation et dolor cupidatat. Qui ipsum mollit sit deserunt ipsum consectetur commodo duis excepteur labore qui.",
      "date": "2018-05-28T16:15:27.118Z"
    },
    {
      "reviewer_name": "Avery",
      "dp": "15",
      "body": "Lorem laborum est ullamco. Minim dolor incididunt magna velit aute officia. Sunt fugiat elit non laborum in incididunt duis excepteur laboris tempor.",
      "date": "2019-02-11T12:55:55.113Z",
      "respond": "Nulla irure et Lorem eiusmod fugiat commodo commodo fugiat labore officia. Magna dolore eiusmod tempor velit enim laboris cillum adipisicing. In consectetur consequat reprehenderit veniam mollit labore dolor. Labore aute exercitation aliqua tempor laborum culpa culpa laborum irure eu aute non excepteur reprehenderit incididunt. Ex mollit occaecat qui. Magna aliquip ipsum ea exercitation consequat nulla exercitation reprehenderit. Incididunt minim amet reprehenderit sunt cillum voluptate velit minim. Ex sunt ad sit cillum nisi laborum aliqua ex eu."
    },
    {
      "reviewer_name": "Tremaine",
      "dp": "29",
      "body": "Consequat adipisicing veniam non laboris nisi laborum aliquip elit. Anim magna ad esse quis consectetur velit irure est irure adipisicing id aliquip commodo. Nostrud consequat laboris cillum tempor cupidatat culpa enim officia. Do commodo elit quis. Reprehenderit consequat consequat cillum excepteur do sunt.",
      "date": "2019-06-23T04:22:40.529Z",
      "respond": "Laboris aliqua anim ex laboris est ad amet ex nisi aliquip cillum enim. Est excepteur ipsum eiusmod ipsum aliqua anim id enim ex dolore ut. Mollit labore et nisi cillum tempor nulla. Eiusmod cupidatat veniam sunt. Aliqua nisi anim et commodo cillum aute tempor magna reprehenderit qui deserunt. Ipsum officia fugiat est et quis do esse ullamco qui deserunt veniam minim amet quis dolor. Amet exercitation deserunt dolor in veniam minim labore ut consectetur. Deserunt voluptate adipisicing commodo duis fugiat adipisicing cupidatat culpa irure excepteur in velit."
    }
  ],
  "_id": "5ef56729b29e8e2ed9307017",
  "paddedId": "001",
  "userDp": "69",
  "userName": "Herbert",
  "avg": "2.05",
  "reviewSize": 7,
  "cleanAvg": 1.0,
  "commAvg": 2.6,
  "accuracyAvg": 1.2,
  "valueAvg": 1.6,
  "locationAvg": 4.0,
  "checkinAvg": 1.9,
  "__v": 0
}