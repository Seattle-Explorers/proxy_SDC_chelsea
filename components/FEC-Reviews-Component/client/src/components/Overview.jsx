import React from 'react';
import styled from 'styled-components';

const OverviewWrapper = styled.div`
  font-size: 1.5em;
  font-weight: bolder;
`

class Overview extends React.Component {
  render() {
    return (
      <OverviewWrapper>
          {this.props.overview.avg} ({this.props.overview.reviewSize} reviews)
      </OverviewWrapper>
    )
  }
}

export default Overview;