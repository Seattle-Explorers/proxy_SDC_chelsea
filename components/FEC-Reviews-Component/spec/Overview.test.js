import 'jsdom-global/register';
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-enzyme';
import 'jest-styled-components';
import { mount } from 'enzyme';

import 'babel-polyfill';

import mockData from '../__mocks__/mockData';
import Overview from '../client/src/components/Overview';

describe('tests for overview component', () => {
  const { reviews, ...rest } = mockData;

  test('should render', () => {
    const component = <Overview overview={rest} />;

    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should display average and reviews count', () => {
    const component = mount(<Overview overview={rest} />);
    const outputText = component.find('div').text();

    expect(outputText).toBe('2.05 (7 reviews)');
  });
});
