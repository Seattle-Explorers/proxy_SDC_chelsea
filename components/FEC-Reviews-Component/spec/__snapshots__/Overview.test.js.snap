// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for overview component should render 1`] = `
.c0 {
  font-size: 1.5em;
  font-weight: bolder;
}

<div
  className="c0"
>
  2.05
   (
  7
   reviews)
</div>
`;
